api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -261,6 +262,16 @@ message AuthorizationResponse {
   // the authorization response itself is successful, i.e. not failed or denied. When this list is
   // *not* set, no additional headers will be added to the client's response on success.
   type.matcher.v3.ListStringMatcher allowed_client_headers_on_success = 4;
+
+  // When this :ref:`list <envoy_api_msg_type.matcher.v3.ListStringMatcher>` is set, authorization
+  // response headers that have a correspondent match will be emitted as dynamic metadata, whose keys
+  // correspond to the name of the matched header and whose values are the respective values. When this
+  // list is *not* set, no additional dynamic metadata will be emitted. This metadata lives in a namespace
+  // specified by the canonical name of extension filter that requires it:
```

Comment:
@lizan @kyessenov I'm little concerned that we've moved to a model where type URL is the canonical name for extension filters but we continue to use these names in the API for other purposes. Do we need some API style guidance on which one to use where here?

CC @envoyproxy/api-shepherds @phlax

Reply:
Yeah, we need guidance what keys to use for metadata (and filter state). It's wild west right now. As long as the name is unique, it should be OK to use any name, but we need to phrase it this way and avoid canonical name.

Reply:
+1 anything we can do here to centralize this in TypedExtensionConfig and have common documentation would be hugely helpful. This is incredibly confusing IMO.

Reply:
Right, this PR uses the same metadata key name that is documented in the proto used by the gRPC authorization server. What can I do to help here, and is it something that should be addressed in this PR or in a followup PR?

Reply:
For now can you potentially just add an example to these docs? It's pretty hard to understand for the casual user. Thank you!

/wait